USFMConverter/UI/Pages/FileView.axaml.cs

```diff
@@ -0,0 +1,236 @@
+using System;
+using Avalonia;
+using Avalonia.Controls;
+using Avalonia.Input;
+using Avalonia.Markup.Xaml;
+using Avalonia.Platform;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Avalonia.Interactivity;
+using NPOI.OpenXmlFormats.Vml.Office;
+using USFMConverter.Core;
+using USFMConverter.Core.Util;
+using USFMConverter.Core.ConstantValue;
+using USFMConverter.UI.Pages.PartialView;
+
+namespace USFMConverter.UI.Pages
+{
+    public class FileView : UserControl
+    {
+        private ProjectReady projectReadySection;
+        private ProjectNotReady projectNotReadySection;
+        
+        private Border dragDropArea;
+        private ListBox filesContainer;
+        private TextBlock selectedCount;
+        private Button browseBtn;
+        private Button convertBtn;
+        private Button removeFileBtn;
+        private Dictionary<FileFormat, FileDialogFilter> fileFilter = new()
+        {
+            [FileFormat.DOCX] = new FileDialogFilter { 
+                Name = "Word Document", 
+                Extensions = new() { "docx" } 
+            },
+            [FileFormat.HTML] = new FileDialogFilter
+            {
+                Name = "HTML Document",
+                Extensions = new() { "html" }
+            }
+        };
+
+        public event EventHandler<RoutedEventArgs> ConvertStart
+        {
+            add
+            {
+                AddHandler(StartConvertEvent, value);
+            }
+            remove
+            {
+                RemoveHandler(StartConvertEvent, value);
+            }
+        }
+
+        public static readonly RoutedEvent<RoutedEventArgs> StartConvertEvent = 
+            RoutedEvent.Register<FileView, RoutedEventArgs>(
+                nameof(ConvertStart), 
+                RoutingStrategies.Bubble
+            );
+
+        public FileView()
+        {
+            InitializeComponent();
+        }
+
+        private void InitializeComponent()
+        {
+            AvaloniaXamlLoader.Load(this);
+
+            SetLinuxText();
+
+            projectReadySection = this.FindControl<ProjectReady>("ProjectReady");
```

Comment:
After a while, I finally understand what is going on here, you shouldn't be reaching into other controls from here, if you're going to do that this should be just one giant control. If you want to split things up you can create custom event handlers for a control that the parent can then hook into. Because currently if the button name changes in a reusable custom control it will break here. The custom control should be self contained and reusable in itself and not rely on things in another component this heavily.

Reply:
I'm happy to get your thought on this. The problem that we had was the Project Read/NotReady user control is a child of FileView user control. And because the visibility of these children is dependent on the number of files in the ListBox. I don't know a better way to toggle the visibility of ProjectReady/NotReady controls without the parent (FileView) invoking it

![image](https://user-images.githubusercontent.com/34975907/124000510-a835eb00-d9a1-11eb-89e4-68805751491c.png)

Reply:
I agree with Reuben.  The FileView should not be reaching into the ProjectReady object.  Better to have a controlling object that can listen for and respond to the events generated by the children.

Reply:
I'm not sure if I know a way to do it properly... Here's what I see:
the problem (screenshot) is that ProjectReady and ListBox are siblings, the file count of ListBox cannot bubble up an event that ProjectReady can capture; the common parent is FileView.

Reply:
For the start convert event, I have moved it up to the main parent ProjectDetailScreen to handle it. Then, the ProjectReady's convert button will raise and bubble the event to FileView, then bubble to ProjectDetailScreen where it will be handled